projects/step-frontend/src/lib/angularjs/js/controllers/grid.js

```diff
@@ -56,60 +56,6 @@ angular
     },
   ])
 
-  .controller('AgentListCtrl', [
-    '$scope',
-    '$interval',
-    '$http',
-    'helpers',
-    function ($scope, $interval, $http, helpers) {
-      $scope.$state = 'agents';
```

Comment:
@dvladir I don't understand why it affects our routing implementation, but I really don't like how we work with $scope in the old version. $scope.$state was kindof used as a parameter to communicate between components. Just dropping it now would probably not work since old components still rely on it, still, it might be worth to take a deeper look how it is used in this case and if there is a nicer way to solve it ihere.

Reply:
update: to be on the safe side, I un-deleted it, without a comment yet

Reply:
I don't think that this is regarding that code, since it will not be executed anymore. It is about also migrating the scope reference or replacing them with an alternative, so uncommenting these lines will probably not be helpful except as a reminder.

We should defenitely make a document where we document all code that we removed without replacing it (LIKE XXXRegistry.register..., $scope attributes etc.) then I can sit together with the original author of the code and check if it is fine. What do you think @dvladir @chillbirdo ?

Reply:
that would be also an option, but it would also mean that when we delete something we would need to write each line that we did not directly replace to that document. Would we add file names and line numbers to it? When you would sit down with the original author of the legacy code, how would you find that line of code again? The file, where it has been has changed a lot since the time of documenting. Or am I thinking in the wrong direction?

Wouldn't it be simpler for everyone if we just do these comments (`// cleanup candidate`) over code blocks and then, when we think that migration is done you sit down with the original author and delete one by one of those commented code blocks and check if somethings breaks and how it could be repaired?

Reply:
I don't like to have so much process-meta-information in the code, but we can discuss this in the next meeting. For my solution you would write something like "Removed $scope.$state = 'agents'" in PR SED-1051. Then we can see it there and know the context. The problem if you just write "cleanup candidate" in the code is that we don't know if you replaced "$scope.$state = 'agents'" with a new solution or not.

Reply:
I quickly went legacy controllers replaced by me:
```
deleted

$scope.$state = 'tokens'; (TokenListCtrl)
stateStorage.push($scope, 'parameters', {}); (ParameterListCtrl)
stateStorage.push($scope, 'resources', {}); (ResourceListCtrl)
stateStorage.push($scope, 'scheduler', {}); (SchedulerCtrl)
stateStorage.push($scope, 'screenconfiguration', {}); (ScreenConfigurationCtrl)

not deleted yet:

$scope.$state = 'adapters'; ( AdapterListCtrl )
```

Reply:
@chillbirdo @dvladir I think we should solve this sooner rather than later and created a discussion point for next Monday. For now I will just merge this. Thank you for listing the problematic changes!